ethereum_history_api/circuits/lib/README.md

```diff
@@ -1,58 +1,78 @@
 
-This circuit uses oracles and therefore you need to run the oracle server before generating proofs.
-Oracle server is located in [packages/noir-ethereum-api](../packages/noir-ethereum-api) and can be run by:
+Here is a list of public functions that you should use it you decide to go the `oracles route`:
 
-```sh
-yarn oracle-server
+```rust
+pub fn get_header(block_number: Field) -> BlockHeaderPartial;
 ```
 
-Then, to generate proofs, run:
+```rust
+pub fn get_account(block_no: Field, address: Address) -> AccountWithinBlock;
+```
 
-```sh
-nargo prove --oracle-resolver http://localhost:5555
+```rust
+pub fn get_account_with_storage(block_number: Field, address: Address, storage_key: Bytes32) -> StorageWithinBlock<1>;
+```
+
+```rust
+pub fn get_receipt<...>(block_number: Field, tx_idx: Field, ...) -> TxReceiptWithinBlock<...>;
 ```
 
-More details on Noir Ethereum history api oracles in [project directory](ethereum_history_api/oracles/) and oracle feature in [noir docs](https://noir-lang.org/docs/how_to/how-to-oracles/#step-3---usage-with-nargo).
+### Without oracles
 
-## Generating verifiers
+If you decide to not use Oracles - you will need to provide all the data manually, but you can still use verifier functions to verify data & proofs.
 
-To generate a Solidity verifier for a package run:
+Here is a list of public functions that you should use it you decide to go the `NO oracles route`:
 
-```sh
-nargo codegen-verifier --package package_name
-``` 
+```rust
+pub fn verify_header(chain_id: Field, block_header_partial: BlockHeaderPartial, block_header_rlp: BlockHeaderRlp)
+```
 
-For example, for `get_account` package run:
-```sh
-nargo codegen-verifier --package get_account
+```rust
+pub fn verify_account(address: Address, account: Account, state_proof: StateProof, state_root: [u8; KEY_LENGTH])
 ```
 
-Note: When run for the first time this step may take several minutes. Also, it is not recommended to use VSCode terminal for this command as it seems to run slowly.
+```rust
+pub fn verify_storage_value(storage_root: Bytes32, proof: StorageProof)
+```
 
-After generating a verifier (`contract/package_name/plonk_vk.sol`) its `verify` function can be used to verify proofs.
+```rust
+pub fn verify_storage_values<N>(storage_root: Bytes32, proofs: [StorageProof; N])
+```
 
-## Testing
+```rust
+pub fn verify_receipt<...>(
+    block_number: Field,
+    tx_idx: Field,
+    receipt: TxReceipt<LOG_NUM, MAX_LOG_DATA_SIZE>,
+    receipt_proof: TxReceiptProof<MAX_RECEIPT_PROOF_LENGTH, MAX_RECEIPT_RLP_LENGTH>,
+    receipt_root: [u8; KEY_LENGTH]
+)
+```
 
-To test Noir circuit files run:
+## Package structure
 
 ```sh
-nargo test
+└── src
```

Comment:
I wonder how is package structure important for developer?

Reply:
Because our docs are not exhaustive and don't provide types & generics description - their main use-case is to aid with the quick decision - is this something for me. But when they actually decide to use the code - they need to find the function and read it's full signature. This repo structure helps with this task. We can remove it when the docs are complete, but I also think - it's helpful for new team members. I like it when open source projects give me some sort of a map, so I don't need to look through every package to find things I'm interested in